src/Plaid/PlaidClient.cs

```diff
@@ -165,6 +167,37 @@ public sealed partial class PlaidClient
 			}
 		}
 
+		public readonly async Task<FileResponse> ParseFileResponseAsync()
+		{
+			// NOTE: We do not dispose the HttpResponseMessage here. We pass this duty onto the FileResponse
+			var response = await Message.ConfigureAwait(false);
+			Logger.LogInformation("Completed file request. Url: {Url}, Status Code: {StatusCode}.", Url, response.StatusCode);
+
+			var status = response.StatusCode;
+			if (!response.IsSuccessStatusCode)
+			{
+				var json = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+				var error = ParseError((int)response.StatusCode, json);
+				var badresult = new FileResponse(status, error);
+				response.Dispose();
+
+				return badresult;
+			}
+
+			IEnumerable<KeyValuePair<string, IEnumerable<string>>> inheaders = response.Headers;
```

Comment:
Is there any reason not to just use a `NameValueCollection` for the headers property on the response?

Reply:
Honestly, I'm no fan of `NameValueCollection`. The simple types have better support in Linq, and are more malleable. Plus the headers themselves are based on simple types, not NVC. If the header objects were NVC, I would have just used it for simplicity.

That said, if you want NVC instead, I can make the change.

Reply:
Man, it's been a long day for me. Went looking up the `.Headers` object and found the netfx version (which is NVC) and didn't check. 🤦🏼‍♂️ 

My biggest intention was to avoid LINQ. While I love LINQ personally, I prefer to avoid it in situations with expectations of high performance. However, I think what you've got is fine for now - this will be a rare API call and with a stream, performance shouldn't be an issue anyway. I'm good with what you've got here then.

Reply:
Ugh, I get killed by netfx showing up first all the time. It's a good thing you mentioned Linq aversion, because I would have done the NVC conversion in Linq. Personally, I go to Linq first whenever possible, and then I'll unroll when the profiler proves it's needed.